src/mame/konami/hexion.cpp

```diff
@@ -231,12 +241,20 @@ void hexion_state::bankswitch_w(uint8_t data)
 	// bits 0-3 select ROM bank
 	m_rombank->set_entry(data & 0x0f);
 
-	// does bit 6 trigger the 052591?
+	// bit 6 triggers the 052591
 	if (data & 0x40)
 	{
-		int bank = m_unkram[0] & 1;
-		memset(m_vram[bank], m_unkram[1], 0x2000);
-		m_bg_tilemap[bank]->mark_all_dirty();
+		uint8_t command = m_pmcram[0];
+		if (command <= 1)
+		{
+			memset(m_vram[command], m_pmcram[1], 0x2000);
+			m_bg_tilemap[command]->mark_all_dirty();
+		}
+		else
+		{
+			memset(m_vram[m_pmcram[4]] + ((m_pmcram[3] << 8) + m_pmcram[2]), 0, 16);
+			m_bg_tilemap[m_pmcram[4]]->mark_all_dirty();
+		}
```

Comment:
You’ve removed the mask on the video RAM bank here – that causes a potential heap smash.  You should also do a mask or bounds check on the offset when doing the 16-byte clear as there’s nothing to stop it smashing memory off the end of the area the way it is now.